src/results.rs

```diff
@@ -0,0 +1,278 @@
+        factory: Option<Bound<RowFactory>>,
+    ) -> PyResult<RowsResult> {
+        let row_col_iterator = Py::new(
+            py,
+            RowColumnIterator::new(
+                self.inner
+                    .clone()
+                    .into_rows_result()
+                    .map_err(|e| PyTypeError::new_err(e.to_string()))?,
+            ),
+        )?;
+
+        let f: Py<RowFactory> = match factory {
+            Some(bound) => bound.unbind(),
+            None => Py::new(py, RowFactory::new())?,
+        };
+
+        Ok(RowsResult {
+            row_col_iterator,
+            factory: f,
+        })
+    }
+}
+#[pyclass]
+pub struct RowsResult {
+    row_col_iterator: Py<RowColumnIterator>,
+    factory: Py<RowFactory>,
+}
+
+#[pymethods]
+impl RowsResult {
+    pub fn __next__(&mut self) -> PyResult<Py<PyAny>> {
+        Python::with_gil(|py| {
+            self.row_col_iterator
+                .borrow_mut(py)
+                .yoked
+                .with_mut_return(|view| view.next_row())?;
+
+            self.factory
+                .call_method1(py, "build", (&self.row_col_iterator.bind(py),))
+        })
+    }
+
+    pub fn __iter__(slf: PyRefMut<'_, Self>) -> PyRefMut<'_, Self> {
+        slf
+    }
+}
+
+#[derive(Yokeable)]
+struct IteratorWrapper<'a> {
+    row_iterator: RawRowIterator<'a, 'a>,
+    column_iterator: ColumnIterator<'a, 'a>,
+}
+impl<'a> IteratorWrapper<'a> {
+    fn next_column(&mut self) -> PyResult<Column> {
+        Python::with_gil(|py| {
+            let raw_col = self
+                .column_iterator
+                .next()
+                .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyStopIteration, _>(""))?
+                .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+            let val = PyDeserializedValue::deserialize_py(raw_col.spec.typ(), raw_col.slice, py)
+                .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+            let column_name = PyString::new(py, raw_col.spec.name()).unbind();
+            let value = val.into_pyobject(py).expect("Can't fail").unbind();
+
+            Ok(Column { column_name, value })
+        })
+    }
+
+    fn next_row(&mut self) -> PyResult<()> {
+        let column_iterator = self
+            .row_iterator
+            .next()
+            .ok_or_else(|| PyErr::new::<pyo3::exceptions::PyStopIteration, _>(""))?
+            .map_err(|e| PyRuntimeError::new_err(e.to_string()))?;
+
+        self.column_iterator = column_iterator;
+        Ok(())
+    }
+}
+
+// NOTE: In method attach_to_cart we cannot move out cart so We cannot use into_inner inside of it.
+// Fields in DeserializedMetadataAndRawRows are private We have to think
+// If it would be better to use method into_inner and construct new RawMetadata struct
+// or to implement some public method in rust driver that would give access to raw_rows.
+struct RawMetadata {
+    metadata: ResultMetadataHolder,
+    rows_count: usize,
+    raw_rows: Bytes,
+}
```

Comment:
:+1: `raw_rows` getter has been recently exposed for Node.js-RS Driver. I'm sure we can expose it for Python-RS as well.